### DISCUSSION TITLE
Несколько интересных замечаний и предложений

### DISCUSSION THREAD

**Discussion by maximgrynykha** [bzick/oh-my-backend#10] [Ideas]
* На этапе [Базы данных](https://github.com/bzick/oh-my-backend#%D1%8D%D1%82%D0%B0%D0%BF-5-%D0%B1%D0%B0%D0%B7%D1%8B-%D0%B4%D0%B0%D0%BD%D0%BD%D1%8B%D1%85) я бы акцентировал внимание на двух основных типах операций над SQL базами данных: __DDL__ (Data Definition Language) и __DML__ (Data Manipulation Language).

* На этапе [Проектирование и разработка](https://github.com/bzick/oh-my-backend#%D1%8D%D1%82%D0%B0%D0%BF-12-%D0%BF%D1%80%D0%BE%D0%B5%D0%BA%D1%82%D0%B8%D1%80%D0%BE%D0%B2%D0%B0%D0%BD%D0%B8%D0%B5-%D0%B8-%D1%80%D0%B0%D0%B7%D1%80%D0%B0%D0%B1%D0%BE%D1%82%D0%BA%D0%B0) в пункте *__типы приложений__* я также среди прочих упомянул бы __MPA-приложения__ (Multi-Page Applications).

* На этапе [Протокол HTTP](https://github.com/bzick/oh-my-backend#%D1%8D%D1%82%D0%B0%D0%BF-6-%D0%BF%D1%80%D0%BE%D1%82%D0%BE%D0%BA%D0%BE%D0%BB-http), в пункте *__web-сервер__* имеет смысл упомянуть про __Apache__ поскольку он является основой __LAMP__ стека и большинства локальных веб-серверов с настройками по умолчанию.

* На этапе [Полнотекстовый поиск](https://github.com/bzick/oh-my-backend#%D1%8D%D1%82%D0%B0%D0%BF-10-%D0%BF%D0%BE%D0%BB%D0%BD%D0%BE%D1%82%D0%B5%D0%BA%D1%81%D1%82%D0%BE%D0%B2%D1%8B%D0%B9-%D0%BF%D0%BE%D0%B8%D1%81%D0%BA) также стоит упомянуть о популярном решении - [__Algolia Search__](https://www.algolia.com/products/search-and-discovery/hosted-search-api/)

* На этапе [Проектирование и разработка](https://github.com/bzick/oh-my-backend#%D1%8D%D1%82%D0%B0%D0%BF-12-%D0%BF%D1%80%D0%BE%D0%B5%D0%BA%D1%82%D0%B8%D1%80%D0%BE%D0%B2%D0%B0%D0%BD%D0%B8%D0%B5-%D0%B8-%D1%80%D0%B0%D0%B7%D1%80%D0%B0%D0%B1%D0%BE%D1%82%D0%BA%D0%B0) в пункт *__архитектурные шаблоны__* считаю важным добавить про __SoA__ (Service Oriented Architecthure), __ADR__ (Action, Domain, Responder), а также про __Event sourcing__.

**Comment by maximgrynykha** [comment]
@bzick

**Comment by bzick** [comment]
1) Вроде Data Manipulation Language и Data Manipulation Language и есть SQL, или Вы хотите уточнить терминологию и ввести общую терминологию что есть DDL, DML, DCL и тд?

**Reply by maximgrynykha** [reply]
Пока-что, не уверен как лучше выделить DDL, DML и DCL в документе. Возможно у вас есть какие-то идеи? Просто считаю правильным упомянуть об существовании, и различии в применении основных операций над базами данных, поскольку этот вопрос встречается и на собеседованиях, а люди впервые увидевшие эти аббревиатуры скорее всего либо перепутают их значения, либо вообще не смогут ответить. Да и по правде говоря, я и сам забыл о существовании DCL.

**Reply by maximgrynykha** [reply]
Хорошая статья с их перечислением и объяснением - https://info-comp.ru/what-is-ddl-dml-dcl-tcl

**Reply by bzick** [reply]
да, добавлю

**Reply by bzick** [reply]
Добавил MPA-приложения, SoA, ADR, Apache, Event sourcing, 
Пока не добавил :
- DDL, DML и DCL так как там надо немного изменить раздел (подвинуть пару пунктов)
- Algolia Search, пока не ясно куда собирать saas решения (скорее всего в сниппет)

**Reply by maximgrynykha** [reply]
@bzick спасибо. Я оставил пару комментариев к вашим предыдущим коммитам, взгляните как будет возможность.

**Comment by bzick** [comment]
2) Algolia Search - проприетарное saas решение, оно не укладывается в тот список. надо создать еще один отдельный, для saas решений.

**Reply by bzick** [reply]
Хотя мне не нравится перечислять проприетарные решения, это будет похоже на рекламу

**Reply by maximgrynykha** [reply]
Да, это платное решение, но оно достаточно часто применяется для реализации поиска. Можно сразу вспомнить реализацию поиска на сайтах: [Laravel](https://laravel.com), [VueJS](https://vuejs.org), [TailwindCSS](https://tailwindcss.com). Но я согласен с тем что возможно стоит отложить внесение Algolia в документ, хотя и не считаю лишним упоминать production-ready сервисы, а то что действительно является рекламой помечать лейблом "на правах рекламы".

**Reply by maximgrynykha** [reply]
И мне нравиться идея о создании некоего списка платных, но проверенных решений в конце каждой темы. Но нужно придумать хорошее название для этого раздела, поскольку не все предложения будут SaaS-решениями.
